applications/app/controllers/CampaignsController.scala

```diff
@@ -0,0 +1,59 @@
+package controllers
+
+import common._
+import conf.Configuration
+import model._
+import play.api.libs.json._
+import play.api.libs.ws.{WSClient, WSResponse}
+import play.api.mvc._
+import java.net.URLEncoder
+
+import scala.concurrent._
+
+class CampaignsController(
+  val controllerComponents: ControllerComponents,
+  wsClient: WSClient,
+)(implicit context: ApplicationContext)
+  extends BaseController with ImplicitControllerExecutionContext with Logging {
+
+  private val root = Configuration.formstack.url + "/form/"
+  private val endpoint = "/submission.json"
+  private val token = Configuration.formstack.editorialOauthToken
+
+  def makeUrlSafe(formId: String): String = URLEncoder.encode(formId, "UTF-8")
+
+  def formSubmit() = Action.async { implicit request =>
+
+    val pageUrl: String = request.headers("referer")
+    val jsonBody: Option[JsValue] = request.body.asJson
+
+
+    jsonBody.map { json =>
+      val formId: String = makeUrlSafe((json \ "formId").as[String])
+
+        sendToFormstack(json, formId).flatMap { res =>
+          if (res.status == 201) {
+            Future.successful(Redirect(pageUrl))
+          }
+          else {
+            log.error(s"Reader contribution to callout was sent to Formstack but not saved correctly: ${res}")
+            Future(BadRequest("Sorry your story couldn't be saved"))
```

Comment:
Instead of returning `BadRequest`, we should probably map the error we receive from FormStack. For example, `5xx` should be mapped to `502`. Most `4xx` should be mapped to `500` because they most likely represent a problem with our code. `429` represents that we've exceeded the maximum number of API calls.

Reply:
This should probably always be a 500 (`InternalServerError`) as the client doesn't need to know more, the important thing is that we log the cause on our side